tests/test_saveload.py

```diff
@@ -237,3 +239,14 @@ def test_saveload_usersyms():
     assert_param_between(result2.params['sigma'], 0.7, 2.1)
     assert_param_between(result2.params['center'], 8.4, 8.6)
     assert_param_between(result2.params['height'], 0.2, 1.0)
+
+
+def test_dumps():
+    """Different fitting methods can yield different attributes in
+    ModelRessult. Check that these can be dumped with out problems."""
+    methods = ['leastsq', 'nelder', 'powell', 'cobyla', 'bfgsb',
+               'differential_evolution',
+               'brute', 'basinhopping', 'ampgo', 'shgo', 'dual_annealing']
+
+    for method in methods:
+        test_saveload_modelresult_roundtrip(method=method)
```

Comment:
instead of adding an extra function here, please use the ```@pytest.mark.parametrize``` feature to run the same test with different methods. So for the "roundtrip" test you can do:


```
@pytest.mark.parametrize("methods", ['leastsq', 'nelder', 'powell', 'cobyla', 'bfgsb', 'differential_evolution', 'brute', 'basinhopping', 'ampgo', 'shgo', 'dual_annealing'])
def test_saveload_modelresult_roundtrip(method)
```

Reply:
I agree this is better and will change it